core/src/main/java/org/kohsuke/stapler/export/Model.java

```diff
@@ -190,6 +190,9 @@ void writeNestedObjectTo(T object, TreePruner pruner, DataWriter writer, Set<? e
                 superBlacklist.add(p.name);
             }
             superModel.writeNestedObjectTo(object, pruner, writer, superBlacklist);
+        } else {
+            writer.name("class");
+            writer.value(object.getClass().getSimpleName());
```

Comment:
@svanoort noted a potential NPE here. The `object` parameter of this method should be marked either `@Nonnull` or `@CheckForNull`, according to an inspection of its callers, and the test case should cover null attribute values (and array/collection elements).